back/engines/commercial/user_custom_fields/app/serializers/user_custom_fields/web_api/v1/user_custom_field_serializer.rb

```diff
@@ -21,6 +21,11 @@ class UserCustomFieldSerializer < ::WebApi::V1::CustomFieldSerializer
         ) do |record, _params|
           record.current_ref_distribution
         end
+
+        has_many :projects, serializer: ::WebApi::V1::ProjectSerializer do |object|
+          project_ids = object.permissions.map { |p| p.permission_scope_id if p.permission_scope_type == 'Project' }
+          Project.find(project_ids)
```

Comment:
I think there are several issues here:
1. This does not list timeline projects, where the permission scope is a phase.
2. I think this will give N+1 query issues. For each field, a query is executed to get the projects. We could write a test to check if there are no N+1 query issues, see e.g. content_image_service_spec.rb:148.
3. As far as I know, `find` only returns one project. To find multiple projects through an array of projects, we normally need to do `.where(id: project_ids)`.

Reply:
On 1. Yes that is indeed a big omission
On 2. I don't think this serialization end point is used anywhere outside of admin and usually with a very small number of fields so thinking N+1 queries should be minimal?
on 3. Yes indeed, my tests are only testing for one - will change